sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -48,6 +49,25 @@ class MSSQLEngineAdapter(
     COMMENT_CREATION_TABLE = CommentCreationTable.UNSUPPORTED
     COMMENT_CREATION_VIEW = CommentCreationView.UNSUPPORTED
     SUPPORTS_REPLACE_TABLE = False
+    SCHEMA_DIFFER = SchemaDiffer(
+        parameterized_type_defaults={
+            exp.DataType.build("DECIMAL", dialect=DIALECT).this: [(18, 0), (0,)],
+            exp.DataType.build("BINARY", dialect=DIALECT).this: [(1,)],
+            exp.DataType.build("VARBINARY", dialect=DIALECT).this: [(1,)],
+            exp.DataType.build("CHAR", dialect=DIALECT).this: [(1,)],
+            exp.DataType.build("VARCHAR", dialect=DIALECT).this: [(1,)],
+            exp.DataType.build("NCHAR", dialect=DIALECT).this: [(1,)],
+            exp.DataType.build("NVARCHAR", dialect=DIALECT).this: [(1,)],
```

Comment:
Are we using 1 intentionally here? It seems like the default for casts is 30:

> When n isn't specified in a data definition or variable declaration statement, the default length is 1. When n isn't specified with the CAST function, the default length is 30.

Reply:
The schema differ ingests two schemas and determines the sequence of ALTERs/DROPs/ADDs needed to convert the types/structure of the former to the latter. We want to ALTER if a precision is increased.

If you tell me a schema contains `VARCHAR` and I need to determine whether we can ALTER to `VARCHAR(29)`, we would only want to say yes if we're certain that's possible without truncating data. 

So I think we want to use 1 as the conservative default since we can't know for sure?

Reply:
Makes sense